src/hooks/useApi.js

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react'
+import { useSetRecoilState } from 'recoil'
+import { searchedPodcastsByTermState } from '../states'
+
+export default function useApi() {
+  const [isQuerying, setIsQuerying] = useState(false)
+  const setSearchedPodcastsByTerm = useSetRecoilState(
+    searchedPodcastsByTermState
+  )
+
+  async function searchPodcastsByTerm() {
+    setIsQuerying(true)
+    const res = await fetch('/api/byTerm/deutschlandfunk')
+    const resData = await res.json()
+    setIsQuerying(false)
+    setSearchedPodcastsByTerm(resData.feeds)
+  }
+
+  const api = {
+    searchPodcastsByTerm,
+  }
+
+  return [isQuerying, api]
```

Comment:
The structure is nice if this hook will grow. As long as you only have one function that gets returned you could think about returning the function directly instead of the object api with a function,  which has to get accessed via `api.funcName`.

Reply:
Yes that is right. I just future proofed that api call for further functions